SPICA.Rendering/Model.cs

```diff
@@ -673,147 +673,159 @@ private void RenderMeshes(IEnumerable<Mesh> Meshes, bool isRenderSelected)
 
             var transform = Transform * this.BaseModel.WorldTransform.ToMatrix4();
 
+            List<Mesh> PendingMeshes = Meshes.ToList();
+
             int index = 0;
-            foreach (Mesh Mesh in Meshes)
+            for (int CurrentPriority = 0; CurrentPriority < 256 && PendingMeshes.Count != 0; CurrentPriority++)
             {
-                int n = Mesh.BaseMesh.NodeIndex;
-
-                if (n < MeshNodeVisibilities.Length && !MeshNodeVisibilities[n])
+                PendingMeshes.RemoveAll(Mesh =>
                 {
-                    continue;
-                }
+                    if (Mesh.BaseMesh.Priority != CurrentPriority)
+                    {
+                        return false;
+                    }
 
-                if (index < MeshIndexVisibilities.Length && !MeshIndexVisibilities[index])
-                {
-                    continue;
-                }
+                    int n = Mesh.BaseMesh.NodeIndex;
 
-                index++;
+                    if (n < MeshNodeVisibilities.Length && !MeshNodeVisibilities[n])
+                    {
+                        return true;
+                    }
 
-                if (!Mesh.BaseMesh.IsVisible)
-                    continue;
+                    if (index < MeshIndexVisibilities.Length && !MeshIndexVisibilities[index])
```

Comment:
Please review the MeshIndexVisibilities logic, I think it's wrong (and always has been wrong due to the existance of multiple mesh lists).

Reply:
I added an Index parameter to Mesh which should solve this.

  ```csharp
if (Mesh.Index < MeshIndexVisibilities.Length && !MeshIndexVisibilities[Mesh.Index])
  ```